src/ddsrt/src/events.c

```diff
@@ -13,12 +13,10 @@
 #include "dds/ddsrt/events.h"
 #include "dds/ddsrt/heap.h"
 
-dds_return_t ddsrt_event_socket_init(ddsrt_event_t* ev, ddsrt_socket_t sock, uint32_t flags) {
-  assert(ev);
-  assert(sock);
-  ev->type = ddsrt_event_type_socket;
+void ddsrt_event_socket_init(ddsrt_event_t* ev, ddsrt_socket_t sock, uint32_t flags) 
+{
+  ev->type = DDSRT_EVENT_TYPE_SOCKET;
   ev->flags = flags;
   ev->data.socket.sock = sock;
   ddsrt_atomic_st32(&ev->triggered, DDSRT_EVENT_FLAG_UNSET);
```

Comment:
This was done after a suggestion by @k0ekk0ek for making it more future (and multi-threaded) proof.
I agreed with his assessment that simultaneous reads/writes could cause for uncertainty of the state being read out, so I implemented this status flag with an atomic.
I do not think that it hurts the performance in any major way, and it will prevent some possible weird errors, so my vote is for leaving it as it is now.